### PR TITLE
Improve scoop

### PR DESCRIPTION
**Pull request by dczulada**
Pull requests into Cypress require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [ ] This pull request describes why these changes were made.
- [ ] Internal ticket for this PR:
- [ ] Internal ticket links to this PR
- [ ] Code diff has been done and been reviewed
- [ ] Tests are included and test edge cases
- [ ] Tests have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

**Comment by dczulada** [comment]
https://oncprojectracking.healthit.gov/support/browse/CYPRESS-2845?jql=project%20%3D%20CYPRESS%20AND%20resolution%20%3D%20Unresolved%20ORDER%20BY%20key%20DESC%2C%20priority%20DESC

File: lib/cypress/scoop_and_filter.rb [hunk 1]
```diff
@@ -20,47 +20,92 @@ def get_non_demographic_category_statuses(measures)
...
+    end
 
-      replace_negated_code_with_valueset(data_element, multi_vs_negation_elements)
-      return if data_element.dataElementCodes.blank?
+    def scoop_and_filter_data_element_fields(data_element)
+      # Iterate through each field to see coded fields include relevant codes
+      data_element.fields.keys.each do |field_name|
+        next if data_element[field_name].nil?
+
+        # Dianoses and Facility Locations are unique because they are arrays
```
Comment (elsaperelli) [inline lib/cypress/scoop_and_filter.rb]:
```suggestion
        # Diagnoses and Facility Locations are unique because they are arrays
```

File: test/unit/lib/patient_zipper_test.rb [hunk 2]
```diff
@@ -72,8 +72,7 @@ class PatientZipperTest < ActiveSupport::TestCase
         doc = Nokogiri::XML(zip_entry.get_input_stream, &:strict)
         doc.root.add_namespace_definition('cda', 'urn:hl7-org:v3')
         doc.root.add_namespace_definition('sdtc', 'urn:hl7-org:sdtc')
-        # There should be 1 entry with encounter ids. The fixture measures do not have PhysicalExamPerformed criteria, so that entry will be filtered out
-        assert_equal 1, doc.xpath('//sdtc:templateId[@root="2.16.840.1.113883.10.20.24.3.150"]').size, 'There should be 1 entry with encounter ids'
+        assert_equal 2, doc.xpath('//sdtc:templateId[@root="2.16.840.1.113883.10.20.24.3.150"]').size, 'There should be 1 entry with encounter ids'
```
Comment (elsaperelli) [inline test/unit/lib/patient_zipper_test.rb]: Should this still say "There should be 1 entry with encounter ids" even though we are doing `assert_equal 2`?

**Review by elsaperelli — CHANGES_REQUESTED** [review]
One typo and one question!

File: test/unit/lib/patient_zipper_test.rb [hunk 3]
```diff
@@ -72,8 +72,7 @@ class PatientZipperTest < ActiveSupport::TestCase
         doc = Nokogiri::XML(zip_entry.get_input_stream, &:strict)
         doc.root.add_namespace_definition('cda', 'urn:hl7-org:v3')
         doc.root.add_namespace_definition('sdtc', 'urn:hl7-org:sdtc')
-        # There should be 1 entry with encounter ids. The fixture measures do not have PhysicalExamPerformed criteria, so that entry will be filtered out
-        assert_equal 1, doc.xpath('//sdtc:templateId[@root="2.16.840.1.113883.10.20.24.3.150"]').size, 'There should be 1 entry with encounter ids'
+        assert_equal 2, doc.xpath('//sdtc:templateId[@root="2.16.840.1.113883.10.20.24.3.150"]').size, 'There should be 2 entry with encounter ids'
```
Comment (elsaperelli) [inline test/unit/lib/patient_zipper_test.rb]:
```suggestion
        assert_equal 2, doc.xpath('//sdtc:templateId[@root="2.16.840.1.113883.10.20.24.3.150"]').size, 'There should be 2 entries with encounter ids'
```

**Review by elsaperelli — CHANGES_REQUESTED** [review]
One more small comment!

**Review by elsaperelli — APPROVED** [review]
